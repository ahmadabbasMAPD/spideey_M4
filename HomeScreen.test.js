import React from 'react'; // Importing the React library
import HomeScreen from './HomeScreen'; // Importing the HomeScreen component from the same directory
import renderer from 'react-test-renderer'; // Importing the renderer from react-test-renderer

jest.mock('expo-font'); // Mocking the expo-font module
jest.mock('expo-asset'); // Mocking the expo-asset module

// Define a test for the HomeScreen component
test('renders PatientCard correctly', () => {
  // Render the HomeScreen component to a JavaScript object and create a snapshot
  const tree = renderer.create(<HomeScreen />).toJSON();
  
  // Check if the snapshot matches the previously saved snapshot
  expect(tree).toMatchSnapshot();
});
