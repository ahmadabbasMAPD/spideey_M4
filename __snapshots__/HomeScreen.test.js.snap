// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PatientCard correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      {
        "borderColor": "#00FF00",
        "borderRadius": 75,
        "borderWidth": 2,
        "height": 150,
        "marginBottom": 20,
        "resizeMode": "cover",
        "width": 150,
      }
    }
  />
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
        },
        {
          "fontSize": 30,
          "fontWeight": "bold",
          "marginTop": 10,
          "textShadowColor": "rgba(0, 0, 0, 0.75)",
          "textShadowOffset": {
            "height": 1,
            "width": -1,
          },
          "textShadowRadius": 10,
        },
      ]
    }
  >
    HealthSync
  </Text>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
        },
        {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginBottom": 50,
        },
      ]
    }
  >
    Seamless Care, Empowering Lives
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#00FF00",
        "borderRadius": 10,
        "marginTop": 20,
        "opacity": 1,
        "padding": 15,
        "width": 200,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000000",
          "fontWeight": "bold",
        }
      }
    >
      Login as a Careprovider
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#00FF00",
        "borderRadius": 10,
        "marginTop": 20,
        "opacity": 1,
        "padding": 15,
        "width": 200,
      }
    }
  >
    <Text
      style={
        {
          "color": "#000000",
          "fontWeight": "bold",
        }
      }
    >
      Login as a User
    </Text>
  </View>
</View>
`;
